metabase/metabase
# Move LDAP toggle to the auth card

Epic https://github.com/metabase/metabase/issues/19428

Changes:
- `ldap-enabled` is shown as a toggle on the authentication option card and is changed as a regular setting. The toggle is shown only when this auth method was successfully configured.
- Adds `Save and enable` and `Save but don't enable` buttons in the ldap form when the auth method is not enabled. Removes `ldap-enabled` toggle from the form.
- Add a ldap test server on CI to test ldap connection. Also adds the corresponding e2e helper to configure ldap.

How to test:
- `docker run -p 389:389 osixia/openldap:1.5.0` run to a test ldap server
- Go to Admin -> Settings -> Authentication -> LDAP and enter correct ldap settings:https://github.com/metabase/metabase/blob/80de5d966a1481dc10e4cd5f80af5b0295e305e3/frontend/test/metabase/scenarios/admin/settings/sso/ldap.cy.spec.js#L83
- There should be 2 buttons - `Save and enable` and `Save but don't enable`. Click `Save but don't enable`. It should successfully validate and save the settings.
- Go to Admin -> Settings -> Authentication. A toggle in the top right corner of the LDAP card should appear. Make sure it's possible to turn it on and off. 
- Stop the ldap server and try to enable LDAP via the toggle. It should fail and there should be a message about it.
- Run the server again, go the form, enter port `389` (to make the form think it's different) and click `Save and enable`. Make sure it saves the settings. Go to the Authentication page and make sure the toggle in the card is enabled.
- Go to the form again. There should be only the `Save settings` button. Set the port to `389` and re-save the form. It should save it successfully. 

<img width="530" alt="Screenshot 2022-09-15 at 22 18 48" src="https://user-images.githubusercontent.com/8542534/190490795-ba6ccd68-3188-45c7-9c30-5ded435d6c43.png">
